### PR TITLE
Add dependency on spree_analytics_trackers

### PR DESCRIPTION
**Pull request by VygovskySergey**

### CODE REVIEW

**Comment by damianlegawiec** [comment]
@VygovskySergey this will fail if someone doesn't have `spree_analytics_trackers` extension installed (in development/production env) - in the `Tracker` decorator we need to check if `Trackers` are defined

File: spec/controllers/spree/products_controller_spec.rb [hunk 1]
```diff
@@ -23,11 +22,10 @@
       product.stores << store_1
     end
 
-    it 'should return 404' do
+    it 'should raise ActiveRecord::RecordNotFound' do
+      skip if Spree.version.to_d < 3.3
```
Comment (damianlegawiec) [inline spec/controllers/spree/products_controller_spec.rb]: @VygovskySergey let's add a comment why we're skipping this for previous versions with a link to 404 spree PR

File: app/models/spree/tracker_decorator.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 Spree::Tracker.class_eval do
   belongs_to :store
 
-  def self.current(domain)
+  def self.current(domain=:google_analytics)
```
Comment (damianlegawiec) [inline app/models/spree/tracker_decorator.rb]: @VygovskySergey this needs to be `current(engine = TRACKING_ENGINES.first, domain)`

see https://github.com/spree-contrib/spree_analytics_trackers/blob/master/app/models/spree/tracker.rb#L12

**Comment by dimon-k** [comment]
@damianlegawiec should we merge this one?

**Pull request merged**